lib/hutch/worker.rb

```diff
@@ -103,5 +104,11 @@ def error_acknowledgements
     private
 
     attr_accessor :setup_procs
+
+    def unique_consumer_tag
+      prefix = Hutch::Config[:consumer_tag_prefix]
+      unique_part = SecureRandom.uuid
+      "#{prefix}-#{unique_part}"
```

Comment:
It is. `SecureRandom` is available on JRuby and produces values of a known fixed length (36 bytes).